plasmapy/particles/exceptions.py

```diff
@@ -1,41 +1,61 @@
 """
-Collection of `Exceptions` and `Warnings` for PlasmaPy particles.
+Collection of exceptions and warnings for `plasmapy.particles`.
 """
 __all__ = [
-    "AtomicError",
-    "AtomicWarning",
+    "ParticleError",
+    "ParticleTypeError",
+    "ParticleValueError",
+    "ParticleWarning",
     "ChargeError",
     "InvalidElementError",
     "InvalidIonError",
     "InvalidIsotopeError",
     "InvalidParticleError",
-    "MissingAtomicDataError",
-    "MissingAtomicDataWarning",
+    "MissingParticleDataError",
+    "MissingParticleDataWarning",
     "UnexpectedParticleError",
 ]
 
 from plasmapy.utils import PlasmaPyError, PlasmaPyWarning
 
 
-class AtomicError(PlasmaPyError):
-    """An exception for errors in the `~plasmapy.particles` subpackage."""
+class ParticleError(PlasmaPyError):
+    """Base exception for errors in the `~plasmapy.particles` subpackage."""
 
     pass
 
 
-class MissingAtomicDataError(AtomicError):
+class ParticleTypeError(ParticleError, TypeError):
+    """
+    An exception for errors in the `~plasmapy.particles` subpackage
+    caused because an incorrect type was provided.
+    """
+
+    pass
```

Comment:
Is this going to be for when a wrong particle type is given or when a wrong argument type is given for some functionality in `plasmapy.particles`?  To me it should be used for the former and the later still raises `TypeError`.

Reply:
Hm...I think I'm going to drop `ParticleTypeError` and `ParticleValueError` from this pull request because I'm not actually implementing them anywhere and it's a bit outside the scope of this PR.  For the cases where the wrong particle category is given, there's already `InvalidIonError`, etc.  We could still create these later on if we end up needing something that isn't covered by the existing exceptions.